src/resolver.ts

```diff
@@ -3130,12 +3138,56 @@ export class Resolver extends DiagnosticEmitter {
               }
             }
             if (!fieldType) break; // did report above
+            if (existingField !== null) {
+              // visibility checks
+              let thisFieldIsPublic = fieldPrototype.isPublic;
+              let existingFieldIsPublic = existingField.isPublic;
```

Comment:
There is also the case of:

```ts
class A {
  protected foo: i32;
}
class B extends A {
  [private | public | protected] foo: i32;
}
```

Since this is inheritance here, I guess `protected` can be treated like `public`, except that a `protected` field cannot shadow a `public` field?